File: app/src/main/java/com/example/pizzeria/DAOS/DAOUsuario.java
package com.example.pizzeria.DAOS;

import com.example.pizzeria.ConexionBD.DBHelper;
import com.example.pizzeria.Recursos.Usuario;

import java.util.ArrayList;

public class DAOUsuario {

    private static ArrayList<Usuario> lista;
    private static DAOUsuario dao;
    private DBHelper dbHelper;

    public static DAOUsuario getInstance(){

        if (dao==null)
            dao=new DAOUsuario();
        if(lista==null)
            lista=new ArrayList<Usuario>();
        return dao;
    }

    public void setDbHelper(DBHelper dbHelper) {
        this.dbHelper = dbHelper;
    }

    public ArrayList<Usuario> getLista(){
        if(lista.size()==0) {
            lista.add(new Usuario("admin", "admin", "correo1"));
            lista.add(new Usuario("Manu", "1234", "correo2"));
        }
        return lista;
    }

    public Usuario getUsuario(Usuario usuario){
        if(DAOUsuario.getInstance().getLista().contains(usuario)){
            return usuario;
        }
        else return null;
    }
    public Boolean agregarUsuario(Usuario usuario){
        dbHelper.insertarUsuario(usuario);
        if(DAOUsuario.getInstance().getUsuario(usuario)==null){
            lista.add(usuario);
            return true;
        }

        return false;
    }

}

File: app/src/main/java/com/example/pizzeria/Kotlin/DAOUsuario.kt
package com.example.pizzeria.Kotlin

import com.example.pizzeria.ConexionBD.DBHelper
import com.example.pizzeria.Recursos.Usuario


class DAOUsuario {
    private var dbHelper: DBHelper? = null
    fun setDbHelper(dbHelper: DBHelper?) {
        this.dbHelper = dbHelper
    }

    val lista: ArrayList<Usuario>?
        get() {
            if (Companion.lista!!.size == 0) {
                Companion.lista!!.add(Usuario("admin", "admin", "correo1"))
                Companion.lista!!.add(Usuario("Manu", "1234", "correo2"))
            }
            return Companion.lista
        }

    fun getUsuario(usuario: Usuario): Usuario? {
        return if (instance!!.lista!!.contains(usuario)) {
            usuario
        } else null
    }

    fun agregarUsuario(usuario: Usuario): Boolean {
        dbHelper!!.insertarUsuario(usuario)
        if (instance!!.getUsuario(usuario) == null) {
            Companion.lista!!.add(usuario)
            return true
        }
        return false
    }

    companion object {
        private var lista: ArrayList<Usuario>? = null
        private var dao: DAOUsuario? = null
        val instance: DAOUsuario?
            get() {
                if (dao == null) dao = DAOUsuario()
                if (lista == null) lista = ArrayList()
                return dao
            }
    }
}
